test/10.database.open.js

```diff
@@ -91,30 +91,38 @@ describe('new Database()', function () {
 		expect(fs.existsSync(util.current())).to.be.true;
 	});
 	util.itUnix('should accept the "timeout" option', function () {
-		this.slow(4000);
+		this.slow(5000);
+		const MaxTimeout = 0x7fffffff
 		const testTimeout = (timeout) => {
 			const db = new Database(util.current(), { timeout });
 			try {
 				const start = Date.now();
 				expect(() => db.exec('BEGIN EXCLUSIVE')).to.throw(Database.SqliteError).with.property('code', 'SQLITE_BUSY');
-				const end = Date.now();
-				expect(end - start).to.be.closeTo(timeout, 200);
+				const elapsedMs = Date.now() - start;
+				// slow macOS boxen on GitHub Actions can time out after 1650ms for a 1000ms timeout (!!)
+				expect(elapsedMs, JSON.stringify({ elapsedMs, timeout })).to.be.gte(timeout).and.lt(Math.max(500, timeout * 2));
 			} finally {
 				db.close();
 			}
 		};
-		const blocker = this.db = new Database(util.next(), { timeout: 0x7fffffff });
+		const blocker = this.db = new Database(util.next(), { timeout: MaxTimeout });
 		blocker.exec('BEGIN EXCLUSIVE');
 		testTimeout(0);
+		// timeouts are rounded to the nearest second by SQLite (!!)
 		testTimeout(1000);
 		blocker.close();
-		expect(() => (this.db = new Database(util.current(), { timeout: undefined }))).to.throw(TypeError);
-		expect(() => (this.db = new Database(util.current(), { timeout: null }))).to.throw(TypeError);
-		expect(() => (this.db = new Database(util.current(), { timeout: NaN }))).to.throw(TypeError);
-		expect(() => (this.db = new Database(util.current(), { timeout: '75' }))).to.throw(TypeError);
-		expect(() => (this.db = new Database(util.current(), { timeout: -1 }))).to.throw(TypeError);
-		expect(() => (this.db = new Database(util.current(), { timeout: 75.01 }))).to.throw(TypeError);
-		expect(() => (this.db = new Database(util.current(), { timeout: 0x80000000 }))).to.throw(RangeError);
+		for (const { timeout, err } of [
```

Comment:
I find this offensively worse to read tbh :smile:

Reply:
Yeah, the first time I saw specs written like this, I found it a bit off, as well! Tabular specs have grown on me, though, to the point where I strongly prefer them. It does allow you to DRY up your spec, ensures the same operation is being performed with all the different inputs, and **most importantly, highlights the input and expected results**.

One of the assertions was failing, and I wanted to add the JSON context to it. One solution would be to extract 7 new `it`'s with specific test names, but Josh doesn't follow the one-test-one-spec rule--so I wanted to add the timeout description to the context, but that caused every expectation to linewrap, which was _really_ effective in hiding what the tests were doing.

Reply:
(That said, if @JoshuaWise would rather me rewrite this diff, I don't really care--I just want the PR to go through)